test/ValueGetterTestHelpers.h

```diff
@@ -0,0 +1,290 @@
+          asNormalizedStringViewUnsafe(content), descriptor)};
+    }
+  };
+  LiteralOrIri literalOrIri =
+      toLiteralOrIri(literalContent, literalDescriptor, isIri);
+  auto literal = std::visit(
+      [&](auto&& g) { return g(literalOrIri, &testContext.context); }, getter);
+  return checkLiteralContentAndDatatype(literal, expectedContent,
+                                        expectedDatatype);
+};
+
+}  // namespace valueGetterTestHelpers
+
+namespace unitVGTestHelpers {
+
+using namespace valueGetterTestHelpers;
+
+// Test turtle for
+const std::string unitTtl = R"(
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+<x> <y> "http://example.com"^^xsd:anyURI, 
+        "http://qudt.org/vocab/unit/M"^^xsd:anyURI, 
+        "http://qudt.org/vocab/unit/KiloM"^^xsd:anyURI, 
+        "http://qudt.org/vocab/unit/MI"^^xsd:anyURI, 
+        "http://qudt.org/vocab/unit/example"^^xsd:anyURI, 
+        "http://qudt.org/vocab/unit/MI", 
+        <http://qudt.org/vocab/unit/M>, 
+        <http://qudt.org/vocab/unit/KiloM>, 
+        <http://qudt.org/vocab/unit/MI>, 
+        "1.5"^^<http://example.com>, 
+        "x".
+  )";
+
+// Helper to test UnitOfMeasurementValueGetter using ValueId input
+inline void checkUnitValueGetterFromId(
+    const std::string& fullLiteralOrIri, UnitOfMeasurement expectedResult,
+    sparqlExpression::detail::UnitOfMeasurementValueGetter getter) {
+  TestContextWithGivenTTl testContext{unitTtl};
+  auto actualResult =
+      getter(testContext.getId(fullLiteralOrIri), &testContext.context);
+  ASSERT_EQ(actualResult, expectedResult);
+};
+
+// Helper to test UnitOfMeasurementValueGetter using ValueId input where the
+// ValueId represents an encoded value
+inline void checkUnitValueGetterFromIdEncodedValue(
+    ValueId id, sparqlExpression::detail::UnitOfMeasurementValueGetter getter) {
+  TestContextWithGivenTTl testContext{unitTtl};
+  ASSERT_EQ(getter(id, &testContext.context), UnitOfMeasurement::UNKNOWN);
+}
+
+// Helper to test UnitOfMeasurementValueGetter using ValueId input
+inline void checkUnitValueGetterFromLiteralOrIri(
+    const std::string& unitIriWithoutBrackets, UnitOfMeasurement expectedResult,
+    sparqlExpression::detail::UnitOfMeasurementValueGetter getter) {
+  TestContextWithGivenTTl testContext{unitTtl};
+
+  using LiteralOrIri = ad_utility::triple_component::LiteralOrIri;
+  using Literal = ad_utility::triple_component::Literal;
+  using Iri = ad_utility::triple_component::Iri;
+
+  auto doTest = [&](const ad_utility::triple_component::LiteralOrIri& litOrIri,
+                    bool expectSuccess) {
+    auto actualResult = getter(litOrIri, &testContext.context);
+    ASSERT_EQ(actualResult,
+              expectSuccess ? expectedResult : UnitOfMeasurement::UNKNOWN);
+  };
+
+  // Test xsd:anyURI literal method
+  auto litTest = [&](const std::string& lit, bool expectSuccess) {
+    doTest(LiteralOrIri{Literal::fromStringRepresentation(lit)}, expectSuccess);
```

Comment:
Consider using `Literal::literalWithoutQuotes` and passing an optional datatype as the second argument, so you avoid building the strings on your own. (I know you didn't write the code)